feat: implement deep link analytics consolidation

NicolasMassart opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR implements the complete deep link analytics consolidation, replacing multiple existing deep link modal events with a single `Deep link Used` event. It aligns with the Extension implementation and includes comprehensive app installation detection using Branch.io parameters.

**Key Changes:**
- **Consolidated Analytics**: Replaced 10+ separate deep link modal events with single `DEEP_LINK_USED` event
- **App Installation Detection**: Implemented Branch.io integration for deferred deep link detection
- **Type System**: Created comprehensive type definitions and utility functions
- **Analytics Integration**: Added consolidated event tracking to universal link handlers
- **Route-Specific Parameters**: Implemented sensitive properties extraction for all route types
- **Comprehensive Testing**: Added 45+ unit tests with full coverage

## **Changelog**

CHANGELOG entry: null

## **Related issues**

Fixes: https://github.com/MetaMask/mobile-planning/issues/2211
https://consensyssoftware.atlassian.net/browse/MCWP-263?atlOrigin=eyJpIjoiMGRhMTdlNjAxMWI2NDg0ZmI0YjI1YjE2MjQ0ODhiODQiLCJwIjoiaiJ9 

## **Manual testing steps**

```gherkin
Feature: Deep Link Analytics Consolidation

  Scenario: user opens app via deep link
    Given the app is installed and user has not disabled deep link modals

    When user clicks a deep link (e.g., metamask://swap?from=ETH&to=USDC)
    Then the app should track a single DEEP_LINK_USED event with proper properties

  Scenario: user opens app via Branch.io deferred deep link
    Given the app is not installed and user clicks a Branch.io link

    When user installs and opens the app
    Then the app should track DEEP_LINK_USED with was_app_installed=false

  Scenario: user interacts with deep link modal
    Given the app is installed and user has not disabled deep link modals

    When user clicks a deep link and sees the modal
    And user clicks "Continue"
    Then the app should track DEEP_LINK_USED with interstitial="accepted"

  Scenario: user dismisses deep link modal
    Given the app is installed and user has not disabled deep link modals

    When user clicks a deep link and sees the modal
    And user dismisses the modal
    Then the app should track DEEP_LINK_USED with interstitial="rejected"

  Scenario: user disables deep link modals
    Given the app is installed and user has disabled deep link modals

    When user clicks a deep link
    Then the app should track DEEP_LINK_USED with interstitial="skipped"
```

> [!NOTE]
> To simulate an deep link for testing:
> - **iOS Simulator:** `xcrun simctl openurl booted "http://link.metamask.io/home"`
> - **Android Emulator:** `adb shell am start -a android.intent.action.VIEW -d 
"http://link.metamask.io/home"`
> - **iOS physical device:** `xcrun devicectl device process launch --device [YOUR DEVICE UUID] --terminate-existing --payload-url "http://link.metamask.io/home" io.metamask.MetaMask`

### **Mobile Browser Deep Link Testing**

Visit all of these links (open this PR page in Safari/Chrome), verify the page text and the Continue button work as expected:

- **Http, Home, unsigned**: http://link.metamask.io/home
- **Home, unsigned**: https://link.metamask.io/home
  **Expected log:**
  ```
   (NOBRIDGE) INFO  TRACK event saved {"event": "Deep link Used", "properties": {"applicationVersion": "7.58.0", "attribution_id": "", "currentBuildNumber": "2349", "deviceBrand": "Apple", "interstitial": "accepted", "operatingSystemVersion": "26.0.1", "pageTitle": "Home", "platform": "ios", "route": "home", "signature": "missing", "target": undefined, "utm_campaign": "", "utm_content": "", "utm_medium": "", "utm_source": "", "utm_term": "", "was_app_installed": true}, "type": "track"}
  ```
- **Home, signed**: https://link.metamask.io/home?sig=mmVgbZk8ucIY8-syEWrtnRvZlrHGQgM7Jl26fgRVjuwX62UVvhUE5nxOCWn0kEbYhZ_P17nwCZBbqJU7rPMx2w
  **Expected log:**
  ```
   (NOBRIDGE) INFO  TRACK event saved {"event": "Deep link Used", "properties": {"applicationVersion": "7.58.0", "attribution_id": "", "currentBuildNumber": "2349", "deviceBrand": "Apple", "interstitial": "accepted", "operatingSystemVersion": "26.0.1", "platform": "ios", "route": "home", "signature": "valid", "target": undefined, "utm_campaign": "", "utm_content": "", "utm_medium": "", "utm_source": "", "utm_term": "", "was_app_installed": true}, "type": "track"}
   (NOBRIDGE) INFO  Sent 1 events
  ```
- **Notifications, unsigned**: https://link.metamask.io/notifications
  **Expected log:**
  ```
   (NOBRIDGE) INFO  TRACK event saved {"event": "Deep link Used", "properties": {"applicationVersion": "7.58.0", "attribution_id": "", "currentBuildNumber": "2349", "deviceBrand": "Apple", "interstitial": "not shown", "operatingSystemVersion": "26.0.1", "platform": "ios", "route": "invalid", "signature": "missing", "target": "https://link.metamask.io/notifications", "utm_campaign": "", "utm_content": "", "utm_medium": "", "utm_source": "", "utm_term": "", "was_app_installed": true}, "type": "track"}
   (NOBRIDGE) INFO  Sent 1 events
   ...
   (NOBRIDGE) INFO  TRACK event saved {"event": "Deep link Used", "properties": {"applicationVersion": "7.58.0", "attribution_id": "", "currentBuildNumber": "2349", "deviceBrand": "Apple", "interstitial": "accepted", "operatingSystemVersion": "26.0.1", "pageTitle": undefined, "platform": "ios", "route": "invalid", "signature": "missing", "target": "https://link.metamask.io/notifications", "utm_campaign": "", "utm_content": "", "utm_medium": "", "utm_source": "", "utm_term": "", "was_app_installed": true}, "type": "track"}
   (NOBRIDGE) INFO  Sent 1 events
  ```
- **Notifications, signed**: https://link.metamask.io/notifications?sig=LBG7hmAHPmEn2H4gofEk5uY3t_PInT3am0fAD5UZrOksM0NyG3aKf0E8hwO9UVjl9FNkf_j59sFHRtDWqbo5sw
  **Expected log:**
  ```
   (NOBRIDGE) INFO  TRACK event saved {"event": "Deep link Used", "properties": {"applicationVersion": "7.58.0", "attribution_id": "", "currentBuildNumber": "2349", "deviceBrand": "Apple", "interstitial": "not shown", "operatingSystemVersion": "26.0.1", "platform": "ios", "route": "invalid", "signature": "missing", "target": "https://link.metamask.io/notifications?sig=LBG7hmAHPmEn2H4gofEk5uY3t_PInT3am0fAD5UZrOksM0NyG3aKf0E8hwO9UVjl9FNkf_j59sFHRtDWqbo5sw", "utm_campaign": "", "utm_content": "", "utm_medium": "", "utm_source": "", "utm_term": "", "was_app_installed": true}, "type": "track"}
   (NOBRIDGE) INFO  Sent 1 events

   (NOBRIDGE) INFO  TRACK event saved {"event": "Deep link Used", "properties": {"applicationVersion": "7.58.0", "attribution_id": "", "currentBuildNumber": "2349", "deviceBrand": "Apple", "interstitial": "accepted", "operatingSystemVersion": "26.0.1", "pageTitle": undefined, "platform": "ios", "route": "invalid", "signature": "missing", "target": "https://link.metamask.io/notifications?sig=LBG7hmAHPmEn2H4gofEk5uY3t_PInT3am0fAD5UZrOksM0NyG3aKf0E8hwO9UVjl9FNkf_j59sFHRtDWqbo5sw", "utm_campaign": "", "utm_content": "", "utm_medium": "", "utm_source": "", "utm_term": "", "was_app_installed": true}, "type": "track"}
  ```
- **Swap, unsigned**: https://link.metamask.io/swap?from=eip155%3A1%2Ferc20%3A0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48&to=eip155%3A1%2Ferc20%3A0xdAC17F958D2ee523a2206206994597C13D831ec7&value=0x38d7ea4c68000
  **Expected log:**
  ```
   (NOBRIDGE) INFO  TRACK event saved {"event": "Deep link Used", "properties": {"applicationVersion": "7.58.0", "attribution_id": "", "currentBuildNumber": "2349", "deviceBrand": "Apple", "interstitial": "not shown", "operatingSystemVersion": "26.0.1", "platform": "ios", "route": "swap", "signature": "missing", "target": undefined, "utm_campaign": "", "utm_content": "", "utm_medium": "", "utm_source": "", "utm_term": "", "was_app_installed": true}, "type": "track"}
   (NOBRIDGE) INFO  TRACK event saved {"event": "Deep link Used", "properties": {"applicationVersion": "7.58.0", "attribution_id": "", "currentBuildNumber": "2349", "deviceBrand": "Apple", "from": "eip155:1/erc20:0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48", "interstitial": "not shown", "operatingSystemVersion": "26.0.1", "platform": "ios", "route": "swap", "signature": "missing", "target": undefined, "to": "eip155:1/erc20:0xdAC17F958D2ee523a2206206994597C13D831ec7", "utm_campaign": "", "utm_content": "", "utm_medium": "", "utm_source": "", "utm_term": "", "was_app_installed": true}, "type": "track"}
   (NOBRIDGE) INFO  Sent 2 events
  ```
- **Swap, signed**: https://link.metamask.io/swap?from=eip155%3A1%2Ferc20%3A0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48&to=eip155%3A1%2Ferc20%3A0xdAC17F958D2ee523a2206206994597C13D831ec7&value=0x38d7ea4c68000&sig=hQVhJ5WyDj9DVcKZsBml8yzAKDRSVLMyVYRcTUzArhrhJWeKqwtnObMa7-1-qzwrw0A-vSi39Rt359yVcjrhCw
- **A 404 page**: https://link.metamask.io/this-is-not-a-real-url
- **Home, invalid signature**: https://link.metamask.io/home?sig=BADgbZk8ucIY8-syEWrtnRvZlrHGQgM7Jl26fgRVjuwX62UVvhUE5nxOCWn0kEbYhZ_P17nwCZBbqJU7rPMx2w

Also make sure the "Do not remind me again" checkbox works (it can be unset in the Security Settings)

## **Screenshots/Recordings**

<!-- Analytics events are not visually observable, but can be verified in logs -->

### **Before**
N/A

### **After**
**Http, Home, unsigned**
<img width="400" alt="Http, Home, unsigned" src="https://github.com/user-attachments/assets/efcefe45-6148-4233-a497-03288a9f90fb" />

**Home, signed**
<img width="400" alt="Home, signed" src="https://github.com/user-attachments/assets/bc548d49-11c5-44bc-abb2-34224d9f1e6e" />

**Notifications, unsigned**
<img width="400" alt="Notifications, unsigned" src="https://github.com/user-attachments/assets/1a28a9c9-821d-4cd7-9dca-cb8a726e961f" />

**Notifications, signed**
<img width="400" alt="Notifications, signed" src="https://github.com/user-attachments/assets/6f36362b-75f8-41a2-b87e-174a73a09630" />

**Swap, unsigned**
Name only event:
<img width="400" alt="Swap, unsigned, normal" src="https://github.com/user-attachments/assets/1932a4fa-be5b-4e33-b00f-baa5570b0f16" />
Anonymized event:
<img width="400" alt="Swap, unsigned, anon" src="https://github.com/user-attachments/assets/4addc72f-0a8d-4d7e-936d-3ae1c5cabb60" />

**Swap, signed**
Name only event:
<img width="400" alt="Swap, signed, normal" src="https://github.com/user-attachments/assets/eaeb5fd2-d9de-4b1e-91ce-89a6d22df029" />

Anonymized event:
<img width="400" alt="Swap, signed, anon" src="https://github.com/user-attachments/assets/fbb8ca0f-afb6-4152-8394-0b13938887ac" />

**A 404 page**
<img width="400" alt="A 404 page" src="https://github.com/user-attachments/assets/5e6be99c-03a5-41a6-a6b3-467cec6bef52" />

**Home, invalid signature**
<img width="400" alt="Home, invalid signature" src="https://github.com/user-attachments/assets/5bc3659e-673c-47b4-a35f-a7c94e994a95" />

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable (45+ unit tests added)
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.



<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Consolidates deep link analytics and routes all tracking through a single `DEEP_LINK_USED` event with Branch.io-based install detection and route-specific sensitive props.
> 
> - Adds deep link analytics utilities (`createDeepLinkUsedEventBuilder`), route/signature/interstitial types, and Branch.io integration with timeout; extracts sensitive params per route
> - Refactors `handleUniversalLink` to build analytics context, track consolidated event (including whitelisted/skipped cases), and make `handleDeepLinkModalDisplay` async; updates whitelists and action typing
> - Removes legacy modal-specific analytics and updates `DeepLinkModal` to pure UI + callbacks; adds optional `onContinue` for invalid/unsupported links
> - Migrates deep link/shared types (`deepLink.types.ts`, `deepLinkAnalytics.types.ts`), updates `extractURLParams` and exports; fixes imports for `isInternalDeepLink`
> - Updates card deeplink handlers to new analytics pipeline (`analytics.trackEvent` + `AnalyticsEventBuilder`)
> - Adds extensive unit tests across handlers/analytics and DeepLink modal; updates BrowserTab/Carousel imports
> - Adds docs: `deeplink-analytics.md` and updates deeplinking readme
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 4e0e70f2da4041e64643be8f50a91fc84fcea495. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

NicolasMassart committed:
feat: implement deep link analytics consolidation

- Add consolidated DEEP_LINK_USED event to replace multiple modal events
- Create comprehensive type definitions and utility functions
- Implement app installation detection using Branch.io parameters
- Add analytics integration to handleUniversalLink.ts
- Create extensive unit tests following TDD principles
- Add TODO comments for Week 2 DeepLinkModal refactoring

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

ngalee8311 commented:
feat/2211_deeplinks_metametrics
files changed

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

# Conflicts:
#	app/core/Analytics/MetaMetrics.events.ts

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
feat: consolidate deep link analytics events into single DEEP_LINK_USED event

- Replace 10 individual deep link modal events with consolidated DEEP_LINK_USED event
- Implement app installation detection using Branch.io parameters
- Add InterstitialState enum with ACCEPTED, REJECTED, NOT_SHOWN, SKIPPED states
- Implement sensitive properties extraction for route-specific parameters
- Update DeepLinkModal component to use consolidated analytics
- Add comprehensive unit tests with 100% coverage on new code
- Fix all ESLint and TypeScript compilation errors
- Replace console.warn with Logger.error for proper error handling
- Add TSDoc documentation to all public functions

NicolasMassart committed:
Merge branch 'feat/2211_deeplinks_metametrics' of github.com:MetaMask/metamask-mobile into feat/2211_deeplinks_metametrics

NicolasMassart committed:
refactor(analytics): remove deprecated deep link events and add consolidated analytics utilities

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
formatting

NicolasMassart committed:
fix type

NicolasMassart committed:
move utils to global utils

NicolasMassart committed:
refactor(analytics): consolidate deep link analytics with MetricsEventBuilder

- Replace custom interfaces with MetricsEventBuilder pattern
- Remove unused DeepLinkUsedEventProperties and SensitiveProperties interfaces
- Update createDeepLinkUsedEvent to return MetricsEventBuilder directly
- Fix DeepLinkModal analytics to use proper MetricsEventBuilder pattern
- Add ModalConfig interface for better type safety
- Update unit tests to work with new MetricsEventBuilder return type
- Simplify property extraction using Record<string, string> type
- Improve code maintainability and reduce interface overhead

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

# Conflicts:
#	app/core/Analytics/MetaMetrics.events.ts
#	app/core/DeeplinkManager/ParseManager/extractURLParams.ts

NicolasMassart committed:
refactor(deeplink): consolidate types into centralized type files

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
fixes import

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
refactor: reduce complexity

NicolasMassart committed:
remove dev logs

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
fix InterstitialState

NicolasMassart committed:
lint and format

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
lint

NicolasMassart committed:
Merge branch 'feat/2211_deeplinks_metametrics' of github.com:MetaMask/metamask-mobile into feat/2211_deeplinks_metametrics

NicolasMassart committed:
lint

NicolasMassart committed:
lint fixes

NicolasMassart committed:
lint

NicolasMassart committed:
refactor(deeplinks): reduce cognitive complexity and improve analytics type handling

NicolasMassart committed:
refactor(analytics): create type alias for URL parameter values

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

# Conflicts:
#	app/components/UI/DeepLinkModal/DeepLinkModal.tsx
#	app/components/UI/DeepLinkModal/types.ts
#	app/core/Analytics/MetaMetrics.events.ts
#	app/core/DeeplinkManager/Handlers/handleRewardsUrl.ts
#	app/core/DeeplinkManager/ParseManager/handleUniversalLink.test.ts
#	app/core/DeeplinkManager/ParseManager/handleUniversalLink.ts

smilingkylan commented on app/core/DeeplinkManager/ParseManager/handleUniversalLink.ts:
```diff
@@ -128,35 +195,66 @@ async function handleUniversalLink({
+    const wasInterstitialShown =
+      (interstitialAction as InterstitialState) ===
+        InterstitialState.ACCEPTED ||
+      (interstitialAction as InterstitialState) === InterstitialState.REJECTED;
```
@NicolasMassart Could you comment on these Cursor alerts?

smilingkylan reviewed (dismissed):
LGTM ✅

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
fix supported actions to be an array

NicolasMassart committed:
fix promise not awaited

smilingkylan reviewed (dismissed):
LGTM ✅

NicolasMassart committed:
fix(deeplinks): await async handleDeepLinkModalDisplay to prevent analytics race condition

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
fix(deeplinks): resolve analytics race condition, signature status, and missing previewToken

NicolasMassart committed:
fix(deeplinks): use context route and fix modal tracking inconsistency

NicolasMassart committed:
fix(deeplinks): track analytics before modal dismissal to prevent race condition

NicolasMassart committed:
fix(deeplinks): fix duplicate analytics tracking and incorrect interstitial state calculation

- fix Duplicate/Triple Analytics Tracking
- fix Missing Analytics for REJECTED Interstitial
- fix Incorrect wasInterstitialShown Calculation
- add unit tests

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
Merge remote-tracking branch 'origin/main' into feat/2211_deeplinks_metametrics

# Conflicts:
#	app/core/DeeplinkManager/ParseManager/handleUniversalLink.ts
#	app/core/DeeplinkManager/handlers/legacy/__tests__/handleUniversalLink.test.ts
#	app/core/DeeplinkManager/handlers/legacy/handleDeepLinkModalDisplay.ts
#	app/core/DeeplinkManager/handlers/legacy/handlePerpsUrl.ts
#	app/core/DeeplinkManager/handlers/legacy/handleRewardsUrl.ts
#	app/core/DeeplinkManager/utils/extractURLParams.ts

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
test(deeplinks): fix unit tests for analytics consolidation feature

- Fix module resolution errors in jest.mock calls
- Update ReduxService mock to allow proper state control
- Fix signature status logic for empty sig parameter
- Set interstitialShown to false initially, update when modal shown
- Fix invalid/unsupported links to only pass onBack callback
- Update test expectations to include analytics context parameter
- Add urlObj setup in signature verification test
- Fix ESLint errors: duplicate imports and variable shadowing

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
fix: lint and tsc

NicolasMassart committed:
Merge branch 'feat/2211_deeplinks_metametrics' of github.com:MetaMask/metamask-mobile into feat/2211_deeplinks_metametrics

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
refactor(deeplinks): remove unused analytics context and extract tracking helper

- Remove unused deepLinkContext parameter from DeepLinkModal types
- Remove deepLinkContext parameter from handleDeepLinkModalDisplay
- Extract duplicate analytics tracking code into trackDeepLinkAnalytics helper
- Update all test expectations to match new function signatures

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
fix(deeplinks): convert SUPPORTED_ACTIONS enum to const object to satisfy Sonar rule

NicolasMassart committed:
refactor(deeplinks): extract nested ternary into if-else statement

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
refactor(deeplinks): migrate analytics from MetaMetrics to new analytics system

- Replace MetaMetrics.getInstance().trackEvent() with analytics.trackEvent()
- Replace MetricsEventBuilder with AnalyticsEventBuilder
- Update test mocks to use new analytics system
- Update JSDoc comments to reference AnalyticsEventBuilder

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
fix(deeplinks): populate branchParams in analytics context for deferred deep link detection

- Fetch Branch.io parameters once in handleUniversalLink and include in all analytics contexts
- Refactor detectAppInstallation to accept optional branchParams to avoid duplicate API calls
- Filter undefined values from analytics properties to comply with AnalyticsEventProperties type
- Add comprehensive tests for Branch.io integration and parameter passing
- Add documentation for deep link analytics system

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
fix(e2e): missing mock on  hyperliquid call

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

# Conflicts:
#	e2e/api-mocking/mock-responses/defaults/index.ts

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
refactor: remove default destination assignment in card handlers

Updated `handleCardHome.ts` and `handleCardOnboarding.ts` to remove the default assignment of `CardDeeplinkDestination.CARD_WELCOME` to the `destination` variable, allowing for more flexible handling of card deep links.

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
fix(deeplinks): prioritize user action over disabled state in interstitial analytics

NicolasMassart committed:
fix(analytics): add SELL route mapping for deep link analytics

Add missing SELL and SELL_CRYPTO action mappings to deep link analytics
route mapping function. Previously, SELL actions were incorrectly
reported as 'invalid' route in analytics. This fix:

- Adds DeepLinkRoute.SELL enum value
- Creates extractSellProperties function for SELL route
- Maps ACTIONS.SELL and ACTIONS.SELL_CRYPTO to DeepLinkRoute.SELL
- Adds 'sell' case to URL route extraction
- Adds comprehensive unit tests for SELL route handling
- Updates documentation to include SELL in route mapping table

Fixes issue where valid SELL deep links were misclassified in analytics.

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
fix(deeplink): track primary button action as accepted for invalid/unsupported links

- Add onContinue callback for invalid/unsupported links that tracks ACCEPTED
- Update DeepLinkModal to call onContinue instead of onBack for primary button
- Update type definitions to allow optional onContinue for invalid/unsupported link types
- Add unit tests to verify correct analytics tracking for both primary and close button actions

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

# Conflicts:
#	app/core/DeeplinkManager/handlers/legacy/__tests__/handleUniversalLink.test.ts
#	app/core/DeeplinkManager/handlers/legacy/handleUniversalLink.ts

NicolasMassart committed:
fix(deeplinks): correctly track interstitial state when modals are disabled

- Add interstitialDisabled check to willShowInterstitial calculation
- Distinguish between modal shown (user accepts/rejects) vs auto-accepted (disabled)
- Ensure interstitialShown is false when modal skipped due to disabled setting
- Maintain ACCEPTED action for auto-accepted private links with disabled modals

NicolasMassart committed:
fix(deeplinks): map all supported actions to routes to prevent URL exposure

NicolasMassart committed:
fix(deeplink): correct interstitialShown analytics for PUBLIC links with disabled setting

NicolasMassart committed:
fix(deeplinks): map valid actions to correct routes in analytics

NicolasMassart committed:
test(deeplinks): fix route mapping expectations for DAPP, WC, and CREATE_ACCOUNT actions

Update test expectations to match implementation. The mapSupportedActionToRoute
function correctly maps ACTIONS.DAPP, ACTIONS.WC, and ACTIONS.CREATE_ACCOUNT to
their respective routes (DeepLinkRoute.DAPP, DeepLinkRoute.WC, DeepLinkRoute.CREATE_ACCOUNT),
not DeepLinkRoute.INVALID as the test previously expected.

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

Cal-L reviewed (commented):
Left a comment

NicolasMassart committed:
refactor(deeplinks): move deeplinks utilities to DeeplinkManager scope

- Move app/util/deeplinks to app/core/DeeplinkManager/util/deeplinks
- Update import paths across consuming files
- Fix determineInterstitialState to return SKIPPED when interstitials disabled
- Update tests and documentation

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

NicolasMassart committed:
Merge branch 'main' into feat/2211_deeplinks_metametrics

Cal-L reviewed (approved):
Lgtm
